api/twitter/twitterApi.py

```diff
@@ -2,168 +2,210 @@
+                time.sleep(10)
+                continue
+            parsed_response = response.json()
+            if parsed_response == []:
+                return parsed_response
+            error_response = self.__check_error(api, parsed_response)
+            if error_response:
+                if error_response == 'continue':
+                    continue
+                else:
+                    print('###Error from Twitter:%s\n###Route requested:%s\n###Params:%s' % (error_response, route, params))
+                    return {}
+            else:
+                return parsed_response
+
+    def __check_error(self, api, response):
+        error_response = ""
+        if 'error' in response:
+            error_response = self.__check_twitter_error_code(
+                response["error"])
+        elif 'errors' in response:
+            error_response = self.__check_twitter_error_code(
+                response["errors"][0]["code"])
+        if error_response == "":
+            return False
+        else:
+            if error_response == "Switch helper":
+                api.__get_user_auth()
+                return "continue"
+            elif error_response == "Invalid user":
+                return "invalid user"
+            elif error_response == "Not authorized":
+                return "invalid user"
+            else:
+                return error_response
+
+    @staticmethod
+    def __check_twitter_error_code(code):
+        return {
+            32: "Switch helper",
+            50: "Invalid user",
+            63: "Invalid user",
+            "Not authorized.": "Not authorized",
+            88: "Switch helper",
+            89: "Switch helper",
+            131: "Internal error"
+        }.get(code, "Unknown error %s" % code)
+
+    def get_user_info(self, uid_list):
+        """Request user information, return a dictionary"""
+        results = {}
+        results['response'] = {}
+        route = "users/show"
+        for id in uid_list:
+            params = {'user_id': id}
+            data = self.__request_twitter(route, params)
+            results['response'][ str(id)] = self.__format_user_info( data)
+        return results
+
+    def get_tweet_info(self, tweet_id):
+        """Request tweet information, return a dictionary"""
+        route = "statuses/lookup"
+        params = {'id': tweet_id}
+        data = self.__request_twitter(route, params)
+        if data != []:
+            return self.__format_tweet_info(data)
+        else:
+            return data
+
+    def get_retweeters(self, tweet_id):
+        """Request the 100 last retweet ids, return them as a list"""
+        route = 'statuses/retweeters/ids'
+        params = { 'id': str(tweet_id)}
+        data = self.__request_twitter(route, params)
+        response = {}
+        response['response'] = data['ids']
+        retweeters = response['response']
+        """change user_ids from num to string"""
```

Comment:
doc-strings shouldn't be used inside function bodies except for the very first line.

```suggestion
        # change user_ids from num to string
```